### PR TITLE
Add "Device Characteristics" request

### PR DESCRIPTION
**Pull request by yklishevich**

### CODE REVIEW

File: FantasmoSDK/Classes/Network/FMApi.swift [hunk 1]
```diff
@@ -165,9 +167,20 @@ class FMApi {
             parameters: params,
             token: token,
             completion: postCompletion,
-            error: postError)
+            error: postError
+        )
+    }
+    
+    func sendDeviceCharacteristicsRequest() {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMApi.swift:174]: As I have explained on Slack, we don't know if this will be another request or merely a field in an existing request. Please revert the FMApi changes. They should not be part of this ticket or PR.

**Review by lucaskuzma — CHANGES_REQUESTED** [review]

File: FantasmoSDK/Classes/Network/FMRestClient.swift [hunk 2]
```diff
@@ -18,33 +18,9 @@ struct FMRestClient {
...
-        var data = Data()
-        data.appendParameters(parameters)
-        data.appendFinalBoundary()
-        
-        Self.post(data: data, with: request, completion: completion, error: error)
-    }
+    // MARK: - Internal methods
 
-    /// Post a query with an image to the CPS server
+    /// Post a query with parameters and an optional image to the CPS server
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMRestClient.swift:23]: This is unrelated to the task at hand.

File: FantasmoSDK/Classes/Network/FMRestClient.swift [hunk 3]
```diff
@@ -55,17 +31,18 @@ struct FMRestClient {
     ///   - error: Error closure
     static func post(_ endpoint: FMApiRouter.ApiEndpoint,
                      parameters: [String : String],
-                     imageData: Data,
+                     imageData: Data? = nil,
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMRestClient.swift:34]: This is unrelated to the task at hand.

File: FantasmoSDK/Classes/Network/FMRestClient.swift [hunk 4]
```diff
@@ -55,17 +31,18 @@ struct FMRestClient {
...
                      completion: RestResult? = nil,
                      error: RestError? = nil) {
-        
         let request = Self.requestForEndpoint(endpoint, token: token)
         log.info(String(describing: request.url), parameters: parameters)
         
         var data = Data()
         data.appendParameters(parameters)
-        data.appendImage(imageData)
+        if let theImageData = imageData {
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMRestClient.swift:43]: This is unrelated to the task at hand.

File: FantasmoSDK/Classes/Network/FMApiRouter.swift [hunk 5]
```diff
@@ -11,6 +11,7 @@ struct FMApiRouter {
     enum ApiEndpoint: String {
         case localize = "image.localize"
         case zoneInRadius = "parking.in.radius"
+        case deviceCharacteristics
```
Comment (lucaskuzma) [inline FantasmoSDK/Classes/Network/FMApiRouter.swift:14]: This is unrelated to the task at hand. And there is no spec suggesting that we need another endpoint.

**Review by lucaskuzma — CHANGES_REQUESTED** [review]

File: FantasmoSDK/Classes/Network/FMApi.swift (hunk 1 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMApi.swift:174]: Ok, can I leave renaming `localize()` -> `sendLocalizationRequest()`

File: FantasmoSDK/Classes/Network/FMRestClient.swift (hunk 2 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMRestClient.swift:23]: This is consequence of removing the redundant `post` method in `FMRestClient`. It was needed to refine the comment to other not removed method.

In my first plan was to implement in `FMApi` fully fledged `sendDeviceCharacteristics()` method and later to take from it obtaining params and remove if it will not be needed. `FMRestClient` related code was projected to quickly copy-past from other methods. And when investigating that intended to be copy-pasted code I noticed problem with code duplication in FMRestClient and decided to quickly fix it as not taking much time rather than create separate task for this. Improving error handling in `FMApi` was going to take significant time and I created separate task for it (https://linear.app/fantasmo/issue/MW-92/improve-error-handling-in-ios-sdk)

So this was the consequence of the quick improvement along the way which was highly likely to be forgotten otherwise.

File: FantasmoSDK/Classes/Network/FMRestClient.swift (hunk 3 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMRestClient.swift:34]: The meaning and reason of this change is explained in this comment - https://github.com/fantasmoio/Fantasmo-iOS-SDK/pull/29#discussion_r670308966

Small problems are noticed when working on tasks and they are quickly forgotten afterwards. Creating new tasks for them is not reasonable, ignoring them is also bad thing. As compromise this small improvements are performed immediately on noticing them.

Do you have any suggestion how it can be done better?

File: FantasmoSDK/Classes/Network/FMRestClient.swift (hunk 4 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMRestClient.swift:43]: Reason is explained in https://github.com/fantasmoio/Fantasmo-iOS-SDK/pull/29#discussion_r670313677

File: FantasmoSDK/Classes/Network/FMApiRouter.swift (hunk 5 above)
Comment (yklishevich) [reply FantasmoSDK/Classes/Network/FMApiRouter.swift:14]: With this I must agree
